src/spikeinterface/curation/train_manual_curation.py

```diff
@@ -0,0 +1,796 @@
+        from sklearn.model_selection import train_test_split
+
+        X_train, X_test, y_train, y_test = train_test_split(
+            self.X, self.y, test_size=self.test_size, random_state=self.seed, stratify=self.y
+        )
+        classifier_instances = [self.get_classifier_instance(clf) for clf in self.classifiers]
+        self._evaluate(
+            self.imputation_strategies,
+            self.scaling_techniques,
+            classifier_instances,
+            X_train,
+            X_test,
+            y_train,
+            y_test,
+            self.search_kwargs,
+        )
+
+    def _get_metrics_for_classification(self, analyzer):
+        """Check if required metrics are present and return a DataFrame of metrics for classification."""
+
+        import pandas as pd
+
+        quality_metrics, template_metrics = try_to_get_metrics_from_analyzer(analyzer)
+
+        # Concatenate the available metrics
+        calculated_metrics = pd.concat([m for m in [quality_metrics, template_metrics] if m is not None], axis=1)
+
+        # Remove any metrics for non-existent units, raise error if no units are present
+        calculated_metrics = calculated_metrics.loc[calculated_metrics.index.isin(analyzer.sorting.get_unit_ids())]
+        if calculated_metrics.shape[0] == 0:
+            raise ValueError("No units present in sorting data")
+
+        return calculated_metrics
+
+    def _load_data_files(self, paths):
+        import pandas as pd
+
+        self.testing_metrics = pd.concat([pd.read_csv(path, index_col=0) for path in paths], axis=0)
+
+    def _evaluate(
+        self, imputation_strategies, scaling_techniques, classifiers, X_train, X_test, y_train, y_test, search_kwargs
+    ):
+        from joblib import Parallel, delayed
+        from sklearn.pipeline import Pipeline
+        import pandas as pd
+
+        results = Parallel(n_jobs=self.n_jobs)(
+            delayed(self._train_and_evaluate)(
+                imputation_strategy, scaler, classifier, X_train, X_test, y_train, y_test, idx, search_kwargs
+            )
+            for idx, (imputation_strategy, scaler, classifier) in enumerate(
+                (imputation_strategy, scaler, classifier)
+                for imputation_strategy in imputation_strategies
+                for scaler in scaling_techniques
+                for classifier in classifiers
+            )
+        )
+
+        test_accuracies, models = zip(*results)
+        self.test_accuracies_df = pd.DataFrame(test_accuracies).sort_values("accuracy", ascending=False)
+
+        best_model_id = int(self.test_accuracies_df.iloc[0]["model_id"])
+        best_model, best_imputer, best_scaler = models[best_model_id]
+
+        best_pipeline = Pipeline(
+            [("imputer", best_imputer), ("scaler", best_scaler), ("classifier", best_model.best_estimator_)]
+        )
+
+        self.best_pipeline = best_pipeline
+
+        if self.folder is not None:
+            self._save()
+
+    def _save(self):
```

Comment:
is this tested in `test_train_manual_curation.py`? Maybe it is implicitly tested through testing other functionality

Reply:
Yeah, this is used implicitly during the testing e.g. in the trainer() fixture in `test_train_manual_curation`.